core/src/main/java/org/elasticsearch/rest/RestController.java

```diff
@@ -112,7 +114,51 @@ public synchronized void registerFilter(RestFilter preProcessor) {
     }
 
     /**
-     * Registers a rest handler to be execute when the provided method and path match the request.
+     * Registers a rest handler to be executed when the provided {@code method} and {@code path} match the request.
+     *
+     * @param method GET, POST, etc.
+     * @param path Path to handle (e.g., "/{index}/{type}/_bulk")
+     * @param handler The handler to actually execute the behavior
+     * @param warning The warning to log and send as a header in the response
+     */
+    public void registerDeprecatedHandler(RestRequest.Method method, String path, RestHandler handler, String warning) {
```

Comment:
> Also, I think that registerHandlerAsDeprecationHandler might be a clearer name since we are really wrapping a handler.

I thought I was getting at that without making it too wordy. Happy to change it though as it does seem weird to have "Deprecated" with everything else being "Deprecation".